### ISSUE TITLE
Uses of `\@tempa` in `\Ref`

### ISSUE THREAD

**Issue by muzimuzhi** [latex3/latex2e#1579]
## Brief outline of the enhancement

`\Ref` is the only kernel macro which uses `\@tempa`. It's ok when `\Ref` was provided by `varioref` package, but when it's moved to the kernel, maybe `\@tempa` can be replaced with `\reserved@a`.

https://github.com/latex3/latex2e/blob/803ec3bae0a2835e8f302308a683718fc39b4f94/base/ltxref.dtx#L549-L557

What to do about its rollback codes?

**Comment by FrankMittelbach** [comment]
yes we could/should change that (don't think it warrants rollback, though - I think we can just alter the 2022 code). Otherwise it would mean splitting the rollback into chunks which is painful and not really that helpful in this case.
